prototype/p2p/errors.go

```diff
@@ -0,0 +1,43 @@
+package p2p
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+)
+
+var (
+	ErrPeerHangUp func(error) error = func(err error) error {
```

Comment:
Why create vars rather than just named functions?

Reply:
To keep the error semantics, in relation to the way other errors are defined. These are error factories.

A convention to declare errors and error factories in the same way. Not a pattern.